examples/cluster/demo_kClustering.py

```diff
@@ -0,0 +1,161 @@
+import torch
+import heat as ht
+
+
+def create_spherical_dataset(
+    num_samples_cluster, radius=1.0, offset=4.0, dtype=ht.float32, random_state=1
+):
+    """
+    Creates k=4 sperical clusters in 3D space along the space-diagonal
+
+    Parameters
+    ----------
+    num_samples_cluster: int
+        Number of samples per cluster. Each process will create n // MPI_WORLD.size elements for each cluster
+    radius: float
+        Radius of the sphere
+    offset: float
+        Shift of the clusters along the axes. The 4 clusters will be positioned centered around c1=(offset, offset,offset),
+        c2=(2*offset,2*offset,2*offset), c3=(-offset, -offset, -offset) and c4=(2*offset, -2*offset, -2*offset)
+    dtype: ht.datatype
+    random_state: int
+        seed of the torch random number generator
+    """
+    # contains num_samples
+    p = ht.MPI_WORLD.size
+    # create k sperical clusters with each n elements per cluster. Each process creates k * n/p elements
+    num_ele = num_samples_cluster // p
```

Comment:
add the modulo remainder here?

Reply:
Outdated after usage of heat instead of torch for data creation